From 8928efb34576eaa4de6f186e46b237e06e8c31ca Mon Sep 17 00:00:00 2001
From: Michael Meffie <mmeffie@sinenomine.net>
Date: Mon, 9 Oct 2017 22:16:09 -0400
Subject: [PATCH 3/3] afsmonitor: remove unused LWP_WaitProcess

Remove the unimplemented once-only flag and the unused LWP_WaitProcess
call.

Reviewed-on: https://gerrit.openafs.org/12745
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit 7c27365ea24aed5787f6fc03f30f6085c78ece51)

Change-Id: I3b61f9fb4f45564304b0e35878d3535a10e31d02
Reviewed-on: https://gerrit.openafs.org/14226
Reviewed-by: Andrew Deason <adeason@sinenomine.net>
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Reviewed-by: Mark Vitale <mvitale@sinenomine.net>
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Stephan Wiesand <stephan.wiesand@desy.de>
---
 src/afsmonitor/afsmonitor.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/afsmonitor/afsmonitor.c b/src/afsmonitor/afsmonitor.c
index e7b303071..a6a9e9741 100644
--- a/src/afsmonitor/afsmonitor.c
+++ b/src/afsmonitor/afsmonitor.c
@@ -56,7 +56,6 @@ int afsmon_debug = 0;		/* debug info to file ? */
 FILE *debugFD;			/* debugging file descriptor */
 static int afsmon_output = 0;	/* output to file ? */
 static int afsmon_detOutput = 0;	/* detailed output ? */
-static int afsmon_onceOnly = 0;	/* probe once only ? (not implemented) */
 int afsmon_probefreq;		/* probe frequency */
 static int wpkg_to_use;		/* graphics package to use */
 static char output_filename[80];	/* output filename */
@@ -3794,8 +3793,6 @@ afsmon_execute(void)
 	}
 
 	FSinitFlags = 0;
-	if (afsmon_onceOnly)	/* option not provided at this time */
-	    FSinitFlags |= XSTAT_FS_INITFLAG_ONE_SHOT;
 
 	if (afsmon_debug) {
 	    fprintf(debugFD, "[ %s ] Calling xstat_fs_Init \n", rn);
@@ -3867,8 +3864,6 @@ afsmon_execute(void)
 	collIDs[num_cm_collections++] = AFSCB_XSTATSCOLL_FULL_PERF_INFO;
 
 	CMinitFlags = 0;
-	if (afsmon_onceOnly)	/* once only ? */
-	    CMinitFlags |= XSTAT_CM_INITFLAG_ONE_SHOT;
 
 	if (afsmon_debug) {
 	    fprintf(debugFD, "[ %s ] Calling xstat_cm_Init \n", rn);
@@ -3892,19 +3887,6 @@ afsmon_execute(void)
 
 
     /* end of process cache manager entries */
-    /* if only one probe was required setup a waiting process for the
-     * termination signal */
-    if (afsmon_onceOnly) {
-	code = LWP_WaitProcess(&terminationEvent);
-	if (code) {
-	    if (afsmon_debug) {
-		fprintf(debugFD, "LWP_WaitProcess() returned error %d\n",
-			code);
-		fflush(debugFD);
-	    }
-	    afsmon_Exit(135);
-	}
-    }
 
     /* start the gtx input server */
     code = (intptr_t)gtx_InputServer(afsmon_win);
-- 
2.30.0

